src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Globalization;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    /// <summary>
+    /// Periodically polls SQL's change table to determine if any new changes have occurred to a user's table.
+    /// </summary>
+    /// <remarks>
+    /// Note that there is no possiblity of SQL injection in the raw queries we generate. All parameters that involve
+    /// inserting data from a user table are sanitized. All other parameters are generated exclusively using information
+    /// about the user table's schema (such as primary key column names), data stored in SQL's internal change table, or
+    /// data stored in our own worker table.
+    /// </remarks>
+    /// <typeparam name="T">A user-defined POCO that represents a row of the user's table</typeparam>
+    internal sealed class SqlTableChangeMonitor<T> : IDisposable
```

Comment:
Please add debug logging to all this stuff like we do in https://github.com/Azure/azure-functions-sql-extension/blob/main/src/SqlAsyncCollector.cs

Reply:
I will log the column names and worker table name in `SqlTriggerListener.cs` so those get logged even if there is a failure before `SqlTableChangeMonitor` instance gets constructed. User table name and function ID are already getting logged in `SqlTriggerListener.cs`.